micro_sam/util.py

```diff
@@ -177,7 +131,6 @@ def _available_devices():
 def get_sam_model(
     model_type: str = _DEFAULT_MODEL,
     device: Optional[str] = None,
-    checkpoint_path: Optional[Union[str, os.PathLike]] = None,
```

Comment:
I think we should keep this. Merging this into `get_custom_sam_model` would be confusing and I don't see a big issue keeping it here. See my main comment for more details.

Reply:
Ok. I have restored the `checkpoint_path` keyword argument. 

I will need you to test this (because all the example scripts involving checkpoint_path rely on data you personally have). The `_get_checkpoint` function is not covered by any of the existing tests, so it's safe to say we're not testing anything involving `checkpoint_path` right now. 
Could you please run:
* `examples/annotator_with_custom_model.py`
* `examples/finetuning/use_finetuned_model.py`
* `examples/use_as_library/instance_segmentation.py`

It would be great if we could upload the data and custom weights for `examples/annotator_with_custom_model.py` to zenodo or similar - or even make a similar example but with smaller 2d data and model weights. Ideally then anybody could run this example as a test.

Reply:
Hi @GenevieveBuckley,
the models are all available via zenodo already. See https://github.com/computational-cell-analytics/micro-sam/blob/master/micro_sam/util.py#L46-L49. (I agree though that we should list this in the doc, I have added that to #249)
 and I changed the examples in #280 so that all the examples use sample data.

> The `_get_checkpoint` function is not covered by any of the existing tests, so it's safe to say we're not testing anything involving `checkpoint_path` right now.

I don't think that's true either since merging #280. [This](https://github.com/computational-cell-analytics/micro-sam/blob/dev/test/test_util.py#L36-L40) now uses a custom checkpoint. But you would need to merge the current `dev` branch in here first to get those tests.

I will still go ahead and test the examples with this branch.